File: app/src/main/java/com/sagoforest/common/da/repositories/interfaces/IRepository.java
package com.sagoforest.common.da.repositories.interfaces;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;

/**
 * Created by andy on 3/9/18.
 */

public interface IRepository<T> {
    Flowable<List<T>> getAll();

    Flowable<T> getItem(int id);

    Completable add(T item);

    Completable update(T item);

    Completable remove(int id);

}

File: app/src/main/java/com/sagoforest/common/da/repositories/interfaces/IRepository.kt
package com.sagoforest.common.da.repositories.interfaces

import io.reactivex.Completable
import io.reactivex.Flowable

/**
 * Created by andy on 3/9/18.
 */

interface IRepository<T> {
    val all: Flowable<List<T>>

    fun getItem(id: Int): Flowable<T>

    fun add(item: T): Completable

    fun update(item: T): Completable

    fun remove(id: Int): Completable

}
